packages/contracts-bedrock/scripts/ChainAssertions.sol

```diff
@@ -140,11 +140,23 @@ library ChainAssertions {
     }
 
     /// @notice Asserts that the OptimismMintableERC20Factory is setup correctly
-    function checkOptimismMintableERC20Factory(Types.ContractSet memory _contracts) internal view {
+    function checkOptimismMintableERC20Factory(
+        Types.ContractSet memory _contracts,
+        bool _isInitialized
+    )
+        internal
+        view
+    {
         console.log("Running chain assertions on the OptimismMintableERC20Factory");
         OptimismMintableERC20Factory factory = OptimismMintableERC20Factory(_contracts.OptimismMintableERC20Factory);
-        require(factory.BRIDGE() == _contracts.L1StandardBridge);
-        require(factory.bridge() == _contracts.L1StandardBridge);
+
+        if (!_isInitialized) {
+            require(factory.BRIDGE() == address(0));
+            require(factory.bridge() == address(0));
+        } else {
+            require(factory.BRIDGE() == _contracts.L1StandardBridge);
+            require(factory.bridge() == _contracts.L1StandardBridge);
+        }
```

Comment:
I agree that this doesn't really match the pattern. The other `check_()` functions have an `isProxy` boolean, why is this one different?

Reply:
`checkOptimismMintableERC20Factory` is called at deployment of OptimismMintableERC20Factory https://github.com/ethereum-optimism/optimism/blob/9b513d7c18048bceeb132c1796cf1975fd332e91/packages/contracts-bedrock/scripts/Deploy.s.sol#L555 however, since at deployment the storage variables are set to null values now, checkOptimismMintableERC20Factory fails because the addresses don't match with dependent contracts. there are two options: removing the call to checkOptimismMintableERC20Factory from deployment i.e., removing this line: https://github.com/ethereum-optimism/optimism/blob/9b513d7c18048bceeb132c1796cf1975fd332e91/packages/contracts-bedrock/scripts/Deploy.s.sol#L555 or making checkOptimismMintableERC20Factory aware of the fact that OptimismMintableERC20Factory isn't initialized. originally, I had gone for the first approach, but this comment made me change my mind (highly recommend reading for context): https://github.com/ethereum-optimism/optimism/pull/8631#discussion_r1428267234

thoughts @maurelian @tynes?